tests/test_modeling_utils.py

```diff
@@ -1171,6 +1172,151 @@ def test_safetensors_torch_from_torch_sharded(self):
             self.assertTrue(torch.equal(p1, p2))
 
 
+@require_torch
+class ModelOnTheFlyConversionTester(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.user = "huggingface-hub-ci"
+        cls.repo_name = f"{cls.user}/test-model-on-the-fly"
+        cls._token = os.getenv("HUGGINGFACE_PRODUCTION_USER_TOKEN", None)
```

Comment:
```suggestion
        cls.user = "huggingface-hub-ci"
        cls.repo_id = f"{cls.user}/test-model-on-the-fly-{uuid4()}"
        cls.token = os.getenv("HUGGINGFACE_PRODUCTION_USER_TOKEN", None)
```

You should be careful of making this repo unique otherwise multiple CIs running in parallel with conflict with each other

(nit) preference for `repo_id` when it's the full id (username + repo name)
(nit2) I would store as token as `cls.token` for consistency with `cls.user`/`cls.repo_id`